pkg/provisioning/import.go

```diff
@@ -0,0 +1,712 @@
+			newConfig:       newConfig,
+			pipelineService: s.pipelineService,
+		}}
+	}
+
+	return nil
+}
+
+func (s *Service) prepareConnectorActions(oldConfig, newConfig config.Connector, pipelineID string) []action {
+	if oldConfig.ID == "" {
+		// no old config, it's a brand new connector
+		return []action{createConnectorAction{
+			cfg:              newConfig,
+			pipelineID:       pipelineID,
+			connectorService: s.connectorService,
+			pluginService:    s.pluginService,
+		}}
+	} else if newConfig.ID == "" {
+		// no new config, it's an old connector that needs to be deleted
+		return []action{deleteConnectorAction{
+			cfg:              oldConfig,
+			pipelineID:       pipelineID,
+			connectorService: s.connectorService,
+			pluginService:    s.pluginService,
+		}}
+	}
+
+	// first compare configs but ignore nested configs
+	opts := []cmp.Option{
+		cmp.Comparer(func(p1, p2 config.Processor) bool { return p1.ID == p2.ID }),
+	}
+	if cmp.Equal(oldConfig, newConfig, opts...) {
+		// configs match, no need to do anything
+		return nil
+	}
+
+	// compare them again but ignore fields that can be updated, if configs
+	// are still different an update is not possible, we have to entirely
+	// recreate the connector
+	opts = []cmp.Option{
+		cmpopts.IgnoreFields(config.Connector{}, "Name", "Settings", "Processors"),
+	}
+	if cmp.Equal(oldConfig, newConfig, opts...) {
+		// only updatable fields don't match, we can update the connector
+		return []action{updateConnectorAction{
+			oldConfig:        oldConfig,
+			newConfig:        newConfig,
+			connectorService: s.connectorService,
+		}}
+	}
+
+	// we have to delete the old connector and create a new one
+	return []action{
+		deleteConnectorAction{
+			cfg:              oldConfig,
+			pipelineID:       pipelineID,
+			connectorService: s.connectorService,
+			pluginService:    s.pluginService,
+		},
+		createConnectorAction{
+			cfg:              newConfig,
+			pipelineID:       pipelineID,
+			connectorService: s.connectorService,
+			pluginService:    s.pluginService,
+		},
+	}
+}
+
+func (s *Service) prepareProcessorActions(oldConfig, newConfig config.Processor, parent processor.Parent) []action {
+	if oldConfig.ID == "" {
+		// no old config, it's a brand new processor
+		return []action{createProcessorAction{
+			cfg:              newConfig,
+			parent:           parent,
+			processorService: s.processorService,
+		}}
+	} else if newConfig.ID == "" {
+		// no new config, it's an old processor that needs to be deleted
+		return []action{deleteProcessorAction{
+			cfg:              oldConfig,
+			parent:           parent,
+			processorService: s.processorService,
+		}}
+	}
+
+	// first compare whole configs
+	if cmp.Equal(oldConfig, newConfig) {
+		// configs match, no need to do anything
+		return nil
+	}
+
+	// compare them again but ignore fields that can be updated, if configs
+	// are still different an update is not possible, we have to entirely
+	// recreate the processor
```

Comment:
Thanks for the comment. :+1: In case we add new fields to the processor configs, how can we get notified that we need to ignore those as well?

Reply:
That's a good question. By default, the fields will be treated as "non-updatable", they will trigger a recreation of the processor unless we add them here. I'm not sure what we could do to make this stand out 🤔 I could try to add a test that changes all fields _except_ the ones that should trigger a recreation (i.e. all updatable fields) and expects that this methods creates an update action. Then this test would start failing as soon as we add another field unless we either add that field to the test or to this code. WDYT?

Reply:
If the test doesn't take too much time to write, I think it's a good idea. Otherwise, I believe leaving a comment in the Config struct is good enough for now.